modules/internal/BytesStringCommon.chpl

```diff
@@ -395,7 +395,7 @@ module BytesStringCommon {
         if r.hasLowBound() &&
            x.byteIndices.boundsCheck(r.lowBound:int) &&
            !isInitialByte(x.byte[r.lowBound:int]) {
-          throw new CodepointSplittingError(
+          throw new CodepointSplitError(
```

Comment:
What happens to a user who was relying on the old name?

```chapel
try {
...
}
catch e: CodepointSplittingError {
...
}
...
```

Yes they get a nice deprecation warning, but if they ignore it then their error handling code is broken. I don't really want to just say "add another config that controls the behavior" but may be required here?

Another option would be to define `CodepointSplitError` as
```chapel
class CodepointSplitError: CodepointSplittingError {
...
}
```
I think this would neatly avoid the problem if it won't causes other issues down the line when we go to remove `CodepointSplittingError`

Reply:
That's a good point.

> Another option would be to define CodepointSplitError as ...

I like this idea, but unfortunately it looks like it causes the deprecation warning to propagate to uses of `codePointSplitError` as well.

Reply:
> I like this idea, but unfortunately it looks like it causes the deprecation warning to propagate to uses of codePointSplitError as well.

I think this could be solved by doing
```chapel
class chpl_CodepointSplittingError: Error {
...
}
type CodepointSplittingError = chpl_CodepointSplittingError;
```

Reply:
Now that I think about, the two classes definitions are the same right?

A simpler solution
```chapel
class chpl_CodepointSplitError: Error {
...
}

@deprecated(...)
type CodepointSplittingError = chpl_CodepointSplitError;

type CodepointSplitError = chpl_CodepointSplitError;
```